### PR TITLE
Editor: Fix block template files query for a post-type

### PR DESCRIPTION
**Pull request by Mamaduka**
Sync ticket for https://github.com/WordPress/gutenberg/pull/61244.

Trac ticket: https://core.trac.wordpress.org/ticket/61110.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/block-template-utils.php [hunk 1]
```diff
@@ -383,12 +388,18 @@ function _get_block_templates_files( $template_type, $query = array() ) {
...
 				if (
 					! $post_type ||
 					( $post_type && isset( $candidate['postTypes'] ) && in_array( $post_type, $candidate['postTypes'], true ) )
 				) {
 					$template_files[ $template_slug ] = $candidate;
 				}
+
+				if ( $post_type && ! isset( $candidate['postTypes'] ) && $is_custom ) {
+					$template_files[ $template_slug ] = $candidate;
+				}
```
Comment (mukeshpanchal27) [inline src/wp-includes/block-template-utils.php]: Can we combine two conditions with OR?

Comment (Mamaduka) [reply src/wp-includes/block-template-utils.php]: I think it's easier to parse this way. I see no reason to sacrifice readability for an extra few lines.

**Review by creativecoder — COMMENTED** [review]
The fix works for me, testing it I see that it matches the results from https://github.com/WordPress/gutenberg/pull/61244

Let's add some comments in the custom-hero-template.html file and/or the tests that this template is intentionally excluded from theme.json, otherwise I might look at this in the future and wonder why it's missing from the `customTemplates` array.

**Comment by Mamaduka** [comment]
Thanks for testing, @creativecoder!

I added the explanatory comment for the data provider.

P.S. Comment support for JSON would have been handy at this moment :)

**Comment by vcanales** [comment]
Hey hey @Mamaduka, are we only missing an approval?

**Comment by Mamaduka** [comment]
@vcanales that’s correct.

**Comment by ellatrix** [comment]
Checking blame to find people to review 😄

**Review by oandregal — APPROVED** [review]

**Pull request closed without merging**